### ISSUE TITLE
Fatal TypeError: m.name.indexOf is not a function

### ISSUE THREAD

**Issue by smendola** [shiwano/typhen#9]
Hi,

I'm attempting to use typhen and typhen-json-schema, and can't get past the starting gate.  This may be user error, but the error message is not giving me much of a clue to get past this.  Any help?

Thanks!
Sal

```
% npm -g list typhen
/usr/lib
└── typhen@1.2.7 


% cat test.d.ts
interface Foo {
  f1:number;
  f2:string;
}

% tsc test.d.ts 

% typhen -V -p typhen-json-schema -d . test.d.ts
Parsing TypeScript files
[18:26:54][DEBUG] Loading the TypeScript files
[18:26:54][DEBUG] Reading: /home/horizon/test/test.d.ts
[18:26:54][DEBUG] Reading dafaultLibFile data
[18:26:55][DEBUG] Compiling the TypeScript files
[18:26:56][DEBUG] Parsing the TypeScript symbols
[18:26:56][DEBUG] Creating Module: module=, name=, declarations=
[18:26:56][DEBUG] Creating Module: module=, name=Intl, declarations=@@__defaultLibFile.d.ts(4091,0)
[18:26:56][DEBUG] Creating Interface: module=Intl, name=CollatorOptions, declarations=@@__defaultLibFile.d.ts(4092,4)
[18:26:56][DEBUG] Creating PrimitiveType: module=, name=, declarations=
[18:26:56][DEBUG] Creating Property: module=Intl, name=usage, declarations=@@__defaultLibFile.d.ts(4093,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=localeMatcher, declarations=@@__defaultLibFile.d.ts(4094,8)
[18:26:56][DEBUG] Creating PrimitiveType: module=, name=, declarations=
[18:26:56][DEBUG] Creating Property: module=Intl, name=numeric, declarations=@@__defaultLibFile.d.ts(4095,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=caseFirst, declarations=@@__defaultLibFile.d.ts(4096,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=sensitivity, declarations=@@__defaultLibFile.d.ts(4097,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=ignorePunctuation, declarations=@@__defaultLibFile.d.ts(4098,8)
[18:26:56][DEBUG] Creating Interface: module=Intl, name=ResolvedCollatorOptions, declarations=@@__defaultLibFile.d.ts(4101,4)
[18:26:56][DEBUG] Creating Property: module=Intl, name=locale, declarations=@@__defaultLibFile.d.ts(4102,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=usage, declarations=@@__defaultLibFile.d.ts(4103,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=sensitivity, declarations=@@__defaultLibFile.d.ts(4104,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=ignorePunctuation, declarations=@@__defaultLibFile.d.ts(4105,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=collation, declarations=@@__defaultLibFile.d.ts(4106,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=caseFirst, declarations=@@__defaultLibFile.d.ts(4107,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=numeric, declarations=@@__defaultLibFile.d.ts(4108,8)
[18:26:56][DEBUG] Creating Interface: module=Intl, name=Collator, declarations=@@__defaultLibFile.d.ts(4111,4),@@__defaultLibFile.d.ts(4115,8)
[18:26:56][DEBUG] Creating Parameter: module=Intl, name=x, declarations=@@__defaultLibFile.d.ts(4112,16)
[18:26:56][DEBUG] Creating Parameter: module=Intl, name=y, declarations=@@__defaultLibFile.d.ts(4112,27)
[18:26:56][DEBUG] Creating PrimitiveType: module=, name=, declarations=
[18:26:56][DEBUG] Creating Signature: module=Intl, name=compare, declarations=@@__defaultLibFile.d.ts(4112,8)
[18:26:56][DEBUG] Creating Method: module=Intl, name=compare, declarations=@@__defaultLibFile.d.ts(4112,8)
[18:26:56][DEBUG] Creating Signature: module=Intl, name=resolvedOptions, declarations=@@__defaultLibFile.d.ts(4113,8)
[18:26:56][DEBUG] Creating Method: module=Intl, name=resolvedOptions, declarations=@@__defaultLibFile.d.ts(4113,8)
[18:26:56][DEBUG] Creating Interface: module=Intl, name=NumberFormatOptions, declarations=@@__defaultLibFile.d.ts(4121,4)
[18:26:56][DEBUG] Creating Property: module=Intl, name=localeMatcher, declarations=@@__defaultLibFile.d.ts(4122,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=style, declarations=@@__defaultLibFile.d.ts(4123,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=currency, declarations=@@__defaultLibFile.d.ts(4124,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=currencyDisplay, declarations=@@__defaultLibFile.d.ts(4125,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=useGrouping, declarations=@@__defaultLibFile.d.ts(4126,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=minimumIntegerDigits, declarations=@@__defaultLibFile.d.ts(4127,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=minimumFractionDigits, declarations=@@__defaultLibFile.d.ts(4128,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=maximumFractionDigits, declarations=@@__defaultLibFile.d.ts(4129,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=minimumSignificantDigits, declarations=@@__defaultLibFile.d.ts(4130,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=maximumSignificantDigits, declarations=@@__defaultLibFile.d.ts(4131,8)
[18:26:56][DEBUG] Creating Interface: module=Intl, name=ResolvedNumberFormatOptions, declarations=@@__defaultLibFile.d.ts(4134,4)
[18:26:56][DEBUG] Creating Property: module=Intl, name=locale, declarations=@@__defaultLibFile.d.ts(4135,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=numberingSystem, declarations=@@__defaultLibFile.d.ts(4136,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=style, declarations=@@__defaultLibFile.d.ts(4137,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=currency, declarations=@@__defaultLibFile.d.ts(4138,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=currencyDisplay, declarations=@@__defaultLibFile.d.ts(4139,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=minimumIntegerDigits, declarations=@@__defaultLibFile.d.ts(4140,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=minimumFractionDigits, declarations=@@__defaultLibFile.d.ts(4141,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=maximumFractionDigits, declarations=@@__defaultLibFile.d.ts(4142,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=minimumSignificantDigits, declarations=@@__defaultLibFile.d.ts(4143,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=maximumSignificantDigits, declarations=@@__defaultLibFile.d.ts(4144,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=useGrouping, declarations=@@__defaultLibFile.d.ts(4145,8)
[18:26:56][DEBUG] Creating Interface: module=Intl, name=NumberFormat, declarations=@@__defaultLibFile.d.ts(4148,4),@@__defaultLibFile.d.ts(4152,8)
[18:26:56][DEBUG] Creating Parameter: module=Intl, name=value, declarations=@@__defaultLibFile.d.ts(4149,15)
[18:26:56][DEBUG] Creating Signature: module=Intl, name=format, declarations=@@__defaultLibFile.d.ts(4149,8)
[18:26:56][DEBUG] Creating Method: module=Intl, name=format, declarations=@@__defaultLibFile.d.ts(4149,8)
[18:26:56][DEBUG] Creating Signature: module=Intl, name=resolvedOptions, declarations=@@__defaultLibFile.d.ts(4150,8)
[18:26:56][DEBUG] Creating Method: module=Intl, name=resolvedOptions, declarations=@@__defaultLibFile.d.ts(4150,8)
[18:26:56][DEBUG] Creating Interface: module=Intl, name=DateTimeFormatOptions, declarations=@@__defaultLibFile.d.ts(4158,4)
[18:26:56][DEBUG] Creating Property: module=Intl, name=localeMatcher, declarations=@@__defaultLibFile.d.ts(4159,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=weekday, declarations=@@__defaultLibFile.d.ts(4160,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=era, declarations=@@__defaultLibFile.d.ts(4161,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=year, declarations=@@__defaultLibFile.d.ts(4162,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=month, declarations=@@__defaultLibFile.d.ts(4163,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=day, declarations=@@__defaultLibFile.d.ts(4164,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=hour, declarations=@@__defaultLibFile.d.ts(4165,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=minute, declarations=@@__defaultLibFile.d.ts(4166,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=second, declarations=@@__defaultLibFile.d.ts(4167,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=timeZoneName, declarations=@@__defaultLibFile.d.ts(4168,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=formatMatcher, declarations=@@__defaultLibFile.d.ts(4169,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=hour12, declarations=@@__defaultLibFile.d.ts(4170,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=timeZone, declarations=@@__defaultLibFile.d.ts(4171,8)
[18:26:56][DEBUG] Creating Interface: module=Intl, name=ResolvedDateTimeFormatOptions, declarations=@@__defaultLibFile.d.ts(4174,4)
[18:26:56][DEBUG] Creating Property: module=Intl, name=locale, declarations=@@__defaultLibFile.d.ts(4175,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=calendar, declarations=@@__defaultLibFile.d.ts(4176,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=numberingSystem, declarations=@@__defaultLibFile.d.ts(4177,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=timeZone, declarations=@@__defaultLibFile.d.ts(4178,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=hour12, declarations=@@__defaultLibFile.d.ts(4179,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=weekday, declarations=@@__defaultLibFile.d.ts(4180,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=era, declarations=@@__defaultLibFile.d.ts(4181,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=year, declarations=@@__defaultLibFile.d.ts(4182,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=month, declarations=@@__defaultLibFile.d.ts(4183,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=day, declarations=@@__defaultLibFile.d.ts(4184,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=hour, declarations=@@__defaultLibFile.d.ts(4185,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=minute, declarations=@@__defaultLibFile.d.ts(4186,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=second, declarations=@@__defaultLibFile.d.ts(4187,8)
[18:26:56][DEBUG] Creating Property: module=Intl, name=timeZoneName, declarations=@@__defaultLibFile.d.ts(4188,8)
[18:26:56][DEBUG] Creating Interface: module=Intl, name=DateTimeFormat, declarations=@@__defaultLibFile.d.ts(4191,4),@@__defaultLibFile.d.ts(4195,8)
[18:26:56][DEBUG] Creating UnionType: module=, name=, declarations=
[18:26:56][DEBUG] Creating Interface: module=, name=Date, declarations=@@__defaultLibFile.d.ts(661,0),@@__defaultLibFile.d.ts(840,14),@@__defaultLibFile.d.ts(4221,0),@@__defaultLibFile.d.ts(19458,0)
[18:26:56][DEBUG] Creating Function: module=, name=, declarations=@@__defaultLibFile.d.ts(19459,16)
[18:26:56][DEBUG] Creating Interface: module=, name=VarDate, declarations=@@__defaultLibFile.d.ts(19452,0)
[18:26:56][DEBUG] Creating Signature: module=, name=, declarations=@@__defaultLibFile.d.ts(19459,16)
[18:26:56][DEBUG] Creating Property: module=, name=getVarDate, declarations=@@__defaultLibFile.d.ts(19459,4)
[18:26:56][DEBUG] Creating Signature: module=, name=toString, declarations=@@__defaultLibFile.d.ts(663,4)
[18:26:56][DEBUG] Creating Method: module=, name=toString, declarations=@@__defaultLibFile.d.ts(663,4)
[18:26:56][DEBUG] Creating Signature: module=, name=toDateString, declarations=@@__defaultLibFile.d.ts(665,4)
[18:26:56][DEBUG] Creating Method: module=, name=toDateString, declarations=@@__defaultLibFile.d.ts(665,4)
[18:26:56][DEBUG] Creating Signature: module=, name=toTimeString, declarations=@@__defaultLibFile.d.ts(667,4)
[18:26:56][DEBUG] Creating Method: module=, name=toTimeString, declarations=@@__defaultLibFile.d.ts(667,4)
[18:26:56][DEBUG] Creating Signature: module=, name=toLocaleString, declarations=@@__defaultLibFile.d.ts(669,4),@@__defaultLibFile.d.ts(4227,4)
[18:26:56][DEBUG] Creating UnionType: module=, name=, declarations=
[18:26:56][DEBUG] Creating Array: module=, name=Array, declarations=@@__defaultLibFile.d.ts(1132,0),@@__defaultLibFile.d.ts(1308,14)
[18:26:56][DEBUG] Creating Parameter: module=, name=locales, declarations=@@__defaultLibFile.d.ts(4227,19)
[18:26:56][DEBUG] Creating Parameter: module=, name=options, declarations=@@__defaultLibFile.d.ts(4227,48)
[18:26:56][DEBUG] Creating Signature: module=, name=toLocaleString, declarations=@@__defaultLibFile.d.ts(669,4),@@__defaultLibFile.d.ts(4227,4)
[18:26:56][DEBUG] Creating Method: module=, name=toLocaleString, declarations=@@__defaultLibFile.d.ts(669,4),@@__defaultLibFile.d.ts(4227,4)
[18:26:56][DEBUG] Creating Signature: module=, name=toLocaleDateString, declarations=@@__defaultLibFile.d.ts(671,4),@@__defaultLibFile.d.ts(4233,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=locales, declarations=@@__defaultLibFile.d.ts(4233,23)
[18:26:56][DEBUG] Creating Parameter: module=, name=options, declarations=@@__defaultLibFile.d.ts(4233,52)
[18:26:56][DEBUG] Creating Signature: module=, name=toLocaleDateString, declarations=@@__defaultLibFile.d.ts(671,4),@@__defaultLibFile.d.ts(4233,4)
[18:26:56][DEBUG] Creating Method: module=, name=toLocaleDateString, declarations=@@__defaultLibFile.d.ts(671,4),@@__defaultLibFile.d.ts(4233,4)
[18:26:56][DEBUG] Creating Signature: module=, name=toLocaleTimeString, declarations=@@__defaultLibFile.d.ts(673,4),@@__defaultLibFile.d.ts(4240,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=locales, declarations=@@__defaultLibFile.d.ts(4240,23)
[18:26:56][DEBUG] Creating Parameter: module=, name=options, declarations=@@__defaultLibFile.d.ts(4240,52)
[18:26:56][DEBUG] Creating Signature: module=, name=toLocaleTimeString, declarations=@@__defaultLibFile.d.ts(673,4),@@__defaultLibFile.d.ts(4240,4)
[18:26:56][DEBUG] Creating Method: module=, name=toLocaleTimeString, declarations=@@__defaultLibFile.d.ts(673,4),@@__defaultLibFile.d.ts(4240,4)
[18:26:56][DEBUG] Creating Signature: module=, name=valueOf, declarations=@@__defaultLibFile.d.ts(675,4)
[18:26:56][DEBUG] Creating Method: module=, name=valueOf, declarations=@@__defaultLibFile.d.ts(675,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getTime, declarations=@@__defaultLibFile.d.ts(677,4)
[18:26:56][DEBUG] Creating Method: module=, name=getTime, declarations=@@__defaultLibFile.d.ts(677,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getFullYear, declarations=@@__defaultLibFile.d.ts(679,4)
[18:26:56][DEBUG] Creating Method: module=, name=getFullYear, declarations=@@__defaultLibFile.d.ts(679,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getUTCFullYear, declarations=@@__defaultLibFile.d.ts(681,4)
[18:26:56][DEBUG] Creating Method: module=, name=getUTCFullYear, declarations=@@__defaultLibFile.d.ts(681,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getMonth, declarations=@@__defaultLibFile.d.ts(683,4)
[18:26:56][DEBUG] Creating Method: module=, name=getMonth, declarations=@@__defaultLibFile.d.ts(683,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getUTCMonth, declarations=@@__defaultLibFile.d.ts(685,4)
[18:26:56][DEBUG] Creating Method: module=, name=getUTCMonth, declarations=@@__defaultLibFile.d.ts(685,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getDate, declarations=@@__defaultLibFile.d.ts(687,4)
[18:26:56][DEBUG] Creating Method: module=, name=getDate, declarations=@@__defaultLibFile.d.ts(687,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getUTCDate, declarations=@@__defaultLibFile.d.ts(689,4)
[18:26:56][DEBUG] Creating Method: module=, name=getUTCDate, declarations=@@__defaultLibFile.d.ts(689,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getDay, declarations=@@__defaultLibFile.d.ts(691,4)
[18:26:56][DEBUG] Creating Method: module=, name=getDay, declarations=@@__defaultLibFile.d.ts(691,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getUTCDay, declarations=@@__defaultLibFile.d.ts(693,4)
[18:26:56][DEBUG] Creating Method: module=, name=getUTCDay, declarations=@@__defaultLibFile.d.ts(693,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getHours, declarations=@@__defaultLibFile.d.ts(695,4)
[18:26:56][DEBUG] Creating Method: module=, name=getHours, declarations=@@__defaultLibFile.d.ts(695,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getUTCHours, declarations=@@__defaultLibFile.d.ts(697,4)
[18:26:56][DEBUG] Creating Method: module=, name=getUTCHours, declarations=@@__defaultLibFile.d.ts(697,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getMinutes, declarations=@@__defaultLibFile.d.ts(699,4)
[18:26:56][DEBUG] Creating Method: module=, name=getMinutes, declarations=@@__defaultLibFile.d.ts(699,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getUTCMinutes, declarations=@@__defaultLibFile.d.ts(701,4)
[18:26:56][DEBUG] Creating Method: module=, name=getUTCMinutes, declarations=@@__defaultLibFile.d.ts(701,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getSeconds, declarations=@@__defaultLibFile.d.ts(703,4)
[18:26:56][DEBUG] Creating Method: module=, name=getSeconds, declarations=@@__defaultLibFile.d.ts(703,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getUTCSeconds, declarations=@@__defaultLibFile.d.ts(705,4)
[18:26:56][DEBUG] Creating Method: module=, name=getUTCSeconds, declarations=@@__defaultLibFile.d.ts(705,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getMilliseconds, declarations=@@__defaultLibFile.d.ts(707,4)
[18:26:56][DEBUG] Creating Method: module=, name=getMilliseconds, declarations=@@__defaultLibFile.d.ts(707,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getUTCMilliseconds, declarations=@@__defaultLibFile.d.ts(709,4)
[18:26:56][DEBUG] Creating Method: module=, name=getUTCMilliseconds, declarations=@@__defaultLibFile.d.ts(709,4)
[18:26:56][DEBUG] Creating Signature: module=, name=getTimezoneOffset, declarations=@@__defaultLibFile.d.ts(711,4)
[18:26:56][DEBUG] Creating Method: module=, name=getTimezoneOffset, declarations=@@__defaultLibFile.d.ts(711,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=time, declarations=@@__defaultLibFile.d.ts(716,12)
[18:26:56][DEBUG] Creating Signature: module=, name=setTime, declarations=@@__defaultLibFile.d.ts(716,4)
[18:26:56][DEBUG] Creating Method: module=, name=setTime, declarations=@@__defaultLibFile.d.ts(716,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=ms, declarations=@@__defaultLibFile.d.ts(721,20)
[18:26:56][DEBUG] Creating Signature: module=, name=setMilliseconds, declarations=@@__defaultLibFile.d.ts(721,4)
[18:26:56][DEBUG] Creating Method: module=, name=setMilliseconds, declarations=@@__defaultLibFile.d.ts(721,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=ms, declarations=@@__defaultLibFile.d.ts(726,23)
[18:26:56][DEBUG] Creating Signature: module=, name=setUTCMilliseconds, declarations=@@__defaultLibFile.d.ts(726,4)
[18:26:56][DEBUG] Creating Method: module=, name=setUTCMilliseconds, declarations=@@__defaultLibFile.d.ts(726,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=sec, declarations=@@__defaultLibFile.d.ts(733,15)
[18:26:56][DEBUG] Creating Parameter: module=, name=ms, declarations=@@__defaultLibFile.d.ts(733,28)
[18:26:56][DEBUG] Creating Signature: module=, name=setSeconds, declarations=@@__defaultLibFile.d.ts(733,4)
[18:26:56][DEBUG] Creating Method: module=, name=setSeconds, declarations=@@__defaultLibFile.d.ts(733,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=sec, declarations=@@__defaultLibFile.d.ts(739,18)
[18:26:56][DEBUG] Creating Parameter: module=, name=ms, declarations=@@__defaultLibFile.d.ts(739,31)
[18:26:56][DEBUG] Creating Signature: module=, name=setUTCSeconds, declarations=@@__defaultLibFile.d.ts(739,4)
[18:26:56][DEBUG] Creating Method: module=, name=setUTCSeconds, declarations=@@__defaultLibFile.d.ts(739,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=min, declarations=@@__defaultLibFile.d.ts(746,15)
[18:26:56][DEBUG] Creating Parameter: module=, name=sec, declarations=@@__defaultLibFile.d.ts(746,28)
[18:26:56][DEBUG] Creating Parameter: module=, name=ms, declarations=@@__defaultLibFile.d.ts(746,42)
[18:26:56][DEBUG] Creating Signature: module=, name=setMinutes, declarations=@@__defaultLibFile.d.ts(746,4)
[18:26:56][DEBUG] Creating Method: module=, name=setMinutes, declarations=@@__defaultLibFile.d.ts(746,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=min, declarations=@@__defaultLibFile.d.ts(753,18)
[18:26:56][DEBUG] Creating Parameter: module=, name=sec, declarations=@@__defaultLibFile.d.ts(753,31)
[18:26:56][DEBUG] Creating Parameter: module=, name=ms, declarations=@@__defaultLibFile.d.ts(753,45)
[18:26:56][DEBUG] Creating Signature: module=, name=setUTCMinutes, declarations=@@__defaultLibFile.d.ts(753,4)
[18:26:56][DEBUG] Creating Method: module=, name=setUTCMinutes, declarations=@@__defaultLibFile.d.ts(753,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=hours, declarations=@@__defaultLibFile.d.ts(761,13)
[18:26:56][DEBUG] Creating Parameter: module=, name=min, declarations=@@__defaultLibFile.d.ts(761,28)
[18:26:56][DEBUG] Creating Parameter: module=, name=sec, declarations=@@__defaultLibFile.d.ts(761,42)
[18:26:56][DEBUG] Creating Parameter: module=, name=ms, declarations=@@__defaultLibFile.d.ts(761,56)
[18:26:56][DEBUG] Creating Signature: module=, name=setHours, declarations=@@__defaultLibFile.d.ts(761,4)
[18:26:56][DEBUG] Creating Method: module=, name=setHours, declarations=@@__defaultLibFile.d.ts(761,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=hours, declarations=@@__defaultLibFile.d.ts(769,16)
[18:26:56][DEBUG] Creating Parameter: module=, name=min, declarations=@@__defaultLibFile.d.ts(769,31)
[18:26:56][DEBUG] Creating Parameter: module=, name=sec, declarations=@@__defaultLibFile.d.ts(769,45)
[18:26:56][DEBUG] Creating Parameter: module=, name=ms, declarations=@@__defaultLibFile.d.ts(769,59)
[18:26:56][DEBUG] Creating Signature: module=, name=setUTCHours, declarations=@@__defaultLibFile.d.ts(769,4)
[18:26:56][DEBUG] Creating Method: module=, name=setUTCHours, declarations=@@__defaultLibFile.d.ts(769,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=date, declarations=@@__defaultLibFile.d.ts(774,12)
[18:26:56][DEBUG] Creating Signature: module=, name=setDate, declarations=@@__defaultLibFile.d.ts(774,4)
[18:26:56][DEBUG] Creating Method: module=, name=setDate, declarations=@@__defaultLibFile.d.ts(774,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=date, declarations=@@__defaultLibFile.d.ts(779,15)
[18:26:56][DEBUG] Creating Signature: module=, name=setUTCDate, declarations=@@__defaultLibFile.d.ts(779,4)
[18:26:56][DEBUG] Creating Method: module=, name=setUTCDate, declarations=@@__defaultLibFile.d.ts(779,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=month, declarations=@@__defaultLibFile.d.ts(785,13)
[18:26:56][DEBUG] Creating Parameter: module=, name=date, declarations=@@__defaultLibFile.d.ts(785,28)
[18:26:56][DEBUG] Creating Signature: module=, name=setMonth, declarations=@@__defaultLibFile.d.ts(785,4)
[18:26:56][DEBUG] Creating Method: module=, name=setMonth, declarations=@@__defaultLibFile.d.ts(785,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=month, declarations=@@__defaultLibFile.d.ts(791,16)
[18:26:56][DEBUG] Creating Parameter: module=, name=date, declarations=@@__defaultLibFile.d.ts(791,31)
[18:26:56][DEBUG] Creating Signature: module=, name=setUTCMonth, declarations=@@__defaultLibFile.d.ts(791,4)
[18:26:56][DEBUG] Creating Method: module=, name=setUTCMonth, declarations=@@__defaultLibFile.d.ts(791,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=year, declarations=@@__defaultLibFile.d.ts(798,16)
[18:26:56][DEBUG] Creating Parameter: module=, name=month, declarations=@@__defaultLibFile.d.ts(798,30)
[18:26:56][DEBUG] Creating Parameter: module=, name=date, declarations=@@__defaultLibFile.d.ts(798,46)
[18:26:56][DEBUG] Creating Signature: module=, name=setFullYear, declarations=@@__defaultLibFile.d.ts(798,4)
[18:26:56][DEBUG] Creating Method: module=, name=setFullYear, declarations=@@__defaultLibFile.d.ts(798,4)
[18:26:56][DEBUG] Creating Parameter: module=, name=year, declarations=@@__defaultLibFile.d.ts(805,19)
[18:26:56][DEBUG] Creating Parameter: module=, name=month, declarations=@@__defaultLibFile.d.ts(805,33)
[18:26:56][DEBUG] Creating Parameter: module=, name=date, declarations=@@__defaultLibFile.d.ts(805,49)
[18:26:56][DEBUG] Creating Signature: module=, name=setUTCFullYear, declarations=@@__defaultLibFile.d.ts(805,4)
[18:26:56][DEBUG] Creating Method: module=, name=setUTCFullYear, declarations=@@__defaultLibFile.d.ts(805,4)
[18:26:56][DEBUG] Creating Signature: module=, name=toUTCString, declarations=@@__defaultLibFile.d.ts(807,4)
[18:26:56][DEBUG] Creating Method: module=, name=toUTCString, declarations=@@__defaultLibFile.d.ts(807,4)
[18:26:56][DEBUG] Creating Signature: module=, name=toISOString, declarations=@@__defaultLibFile.d.ts(809,4)
[18:26:56][DEBUG] Creating Method: module=, name=toISOString, declarations=@@__defaultLibFile.d.ts(809,4)
[18:26:56][DEBUG] Creating PrimitiveType: module=, name=, declarations=
[18:26:56][DEBUG] Creating Parameter: module=, name=key, declarations=@@__defaultLibFile.d.ts(811,11)
[18:26:56][DEBUG] Creating Signature: module=, name=toJSON, declarations=@@__defaultLibFile.d.ts(811,4)
[18:26:56][DEBUG] Creating Method: module=, name=toJSON, declarations=@@__defaultLibFile.d.ts(811,4)
Fatal TypeError: m.name.indexOf is not a function
    at /usr/lib/node_modules/typhen/lib/typescript_parser.js:550:70
    at Array.filter (native)
    at TypeScriptParser.parseGenericType (/usr/lib/node_modules/typhen/lib/typescript_parser.js:550:34)
    at TypeScriptParser.parseType (/usr/lib/node_modules/typhen/lib/typescript_parser.js:233:26)
    at /usr/lib/node_modules/typhen/lib/typescript_parser.js:695:64
    at Array.map (native)
    at TypeScriptParser.parseUnionType (/usr/lib/node_modules/typhen/lib/typescript_parser.js:695:32)
    at TypeScriptParser.parseType (/usr/lib/node_modules/typhen/lib/typescript_parser.js:195:22)
    at TypeScriptParser.parseParameter (/usr/lib/node_modules/typhen/lib/typescript_parser.js:797:34)
    at /usr/lib/node_modules/typhen/lib/typescript_parser.js:773:84
    at Array.map (native)
    at TypeScriptParser.parseSignature (/usr/lib/node_modules/typhen/lib/typescript_parser.js:773:52)
    at /usr/lib/node_modules/typhen/lib/typescript_parser.js:762:87
    at Array.map (native)
    at TypeScriptParser.parseMethod (/usr/lib/node_modules/typhen/lib/typescript_parser.js:762:55)
    at /usr/lib/node_modules/typhen/lib/typescript_parser.js:549:46



```

**Comment by shiwano** [comment]
Thank you for your reporting! :smiley:
I've fixed the issue, and published new version as v1.2.8.

**Comment by smendola** [comment]
Thanks! and your reward is another issue :-)
